mayastor/src/bdev/dev.rs

```diff
@@ -98,3 +96,30 @@ fn reject_unknown_parameters(
         Ok(())
     }
 }
+
+// Lookup up a block device via its symbolic name.
+pub fn device_lookup(name: &str) -> Option<Box<dyn BlockDevice>> {
+    debug!("Looking up device by name: {}", name);
+    // First try to lookup NVMF devices, then try to lookup SPDK native devices.
+    nvmx::lookup_by_name(name).or_else(|| SpdkBlockDevice::lookup_by_name(name))
+}
+
+#[instrument]
+pub async fn device_create(uri: &str) -> Result<String, NexusBdevError> {
+    Uri::parse(uri)?.create().await
+}
+
+#[instrument]
+pub async fn device_destroy(uri: &str) -> Result<(), NexusBdevError> {
+    Uri::parse(uri)?.destroy().await
+}
+
+#[instrument]
+pub fn device_open(
+    name: &str,
+    read_write: bool,
+) -> Result<Box<dyn BlockDeviceDescriptor>, CoreError> {
+    // First try to open NVMF devices, then try to lookup SPDK native devices.
+    nvmx::open_by_name(name, read_write)
+        .or_else(|_| SpdkBlockDevice::open_by_name(name, read_write))
+}
```

Comment:
Is it worth generating a log message here indicating which one "wins"

Reply:
We already instrument the input/output from this function, so I presume its fine.

Reply:
I dont see value in that...